### DISCUSSION TITLE
any challenge or potential risk with using osconfig module

### DISCUSSION THREAD

**Discussion by yiyadk** [Azure/azure-osconfig#460] [General]
what's the minimum requirement for deploying the osconfig module in a remote VM/computer? 

Is it possible the host VM/device disable any network or system configurations, which cause problems of using azure-osconfig? 

Thanks.

**Comment by MatthewMWR** [comment]
I'll start with a general answer. Let me know if you want to get into more specifics of your scenario.

Using system management tech (whether it be Intune with PCs, Ansible with servers, IoT Hub + osconfig with edge devices, etc.) you generally have the ability as a sysadmin to make changes to the remote system which would render it unreachable and unmanageable. 

For example, if you use any of these to shut down (rather than reboot) a remote system then you might lose access to (until you can take some undo action such as physically restarting it). Similarly, you could use any of these systems to create firewall rules on the endpoint  that block connectivity to the management system, again rendering it unreachable until some undo action can be taken such as using a local shell to remove the firewall rules.

Is this the information you were looking for?
